CMakeLists.txt

```diff
@@ -119,91 +128,151 @@ set(BOOST_MODULES_TO_FIND filesystem system)
-		                      set(BOOST_TOOLSET "msvc")
-		                  endif()
-		              else()
-		                  set(BOOST_TOOLSET "msvc")
-		              endif()
-		          elseif(CMAKE_HOST_SYSTEM_NAME STREQUAL "Linux")
-		              if(CMAKE_CXX_COMPILER_ID STREQUAL "Clang")
-		                  set(BOOST_TOOLSET "clang")
-		              elseif(CMAKE_CXX_COMPILER_ID STREQUAL "GNU")
-		                  set(BOOST_TOOLSET "gcc")
-		              endif()
-		          endif()
-
-		          # Determine b2 executable name depending on host system
-		          if(CMAKE_HOST_SYSTEM_NAME STREQUAL "Windows")
-		              set(b2_EXECUTABLE_NAME "b2.exe")
-		          else()
-		              set(b2_EXECUTABLE_NAME "b2")
-		          endif()
-
-		          # Find b2
-		          find_program(b2 NAME ${b2_EXECUTABLE_NAME} PATHS ${BOOST_SOURCEDIR} NO_CACHE NO_DEFAULT_PATH)
-
-		          # Compile b2 if it isn't found
-		          if(b2 STREQUAL "b2-NOTFOUND")
-		              message(STATUS "The b2 buildsystem executable hasn't been found in your system. Compiling b2...")
-		              b2_build()
-
-		              # Find b2 again after compilation
-		              find_program(b2 NAME ${b2_EXECUTABLE_NAME} PATHS ${BOOST_SOURCEDIR} NO_CACHE NO_DEFAULT_PATH REQUIRED)
-		          endif()
-
-		          # Compile Boost
-		          message(STATUS "Compiling Boost...")
-		          boost_compile()
-
-		          # Find Boost again
-		          find_package(Boost COMPONENTS ${BOOST_MODULES_TO_FIND} PATHS ${BOOST_INSTALLDIR} NO_DEFAULT_PATH REQUIRED)
-		      else()
-		          message(STATUS "Custom Boost distribution found in ${BOOST_INSTALLDIR}")
-		      endif()
-		  endif()
-	else()
-		  # Try to find Boost on the system and CMake's default paths
-		  set(Boost_USE_STATIC_LIBS ON)
-		  find_package(Boost COMPONENTS ${BOOST_MODULES_TO_FIND} REQUIRED)
-	endif()
+    # Windows (Specifically MSVC, but it technically applies to all builds regardless of compiler / OS...) doesn't handle
+    # symbol redefinitions gracefully.
+    #
+    # In the case of Boost, BoostConfig.cmake doesn't use CMake's generator expressions when defining it's link libraries,
+    # instead adding the found definitions to the project's link dependencies directly. With no way to perform library detection only.
+    # Worse, get_boost() and boost_compile() runs during configure so we don't have a build configuration yet, and therefore do not know what
+    # build of Boost (Release or Debug) to use at that point. This means that calling find_package(Boost ...) more than once can result
+    # in a project that has both the Debug and Release libraries added as dependencies. (Which will cause a link failure under MSVC.
+    # As the libraries are not fully ABI-compatible with each other.)
+    #
+    # TL;DR: Fix Boost by not calling find_package(Boost ...) if it's already been found once.
+    if (NOT VITA3K_Boost_FOUND)
```

Comment:
```suggestion
    if (NOT TARGET Boost::boost)
```

Reply:
As I've stated in the long post below, this won't work.

Boost already tries to do this itself in it's boost_*-config.cmake files as the very first functional statement, and again in the various variant files (for me at line 74 in both). In all cases Boost's own attempts fail.

Sadly, that means we have to keep track of this ourselves.